PhysicsTools/SelectorUtils/interface/VersionedIdProducer.h

```diff
@@ -132,9 +132,24 @@ template< class PhysicsObjectPtr , class SelectorType >
 void VersionedIdProducer<PhysicsObjectPtr,SelectorType>::
 produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
   constexpr char bitmap_label[] = "Bitmap";
-  
+   
   edm::Handle<Collection> physicsObjectsHandle;
   iEvent.getByToken(physicsObjectSrc_,physicsObjectsHandle);
+  
+  //S. Harper fix for a problem
+  //VID must be the one stop shop for all e/gamma ID calculations
+  //we need to access E/gamma ID in DQM
+  //reco runs VID only on pat::Electrons, we can not use pat::Electrons in DQM
+  //therefore DQM has to re-run VID to get the ValueMaps for reco::GsfElectrons
+  //DQM may run on things without gedGsfElectrons
+  //therefore VID must gracefully fail in this case (or not be run but this is difficult)
+  //hence it now just exits if we dont have a valid collection
+  //open to better suggestions...
+  if(!physicsObjectsHandle.isValid()){
+    edm::LogWarning("InvalidCollection")<<
+      "physics objects collection is invalid, no VID info will be written";
+    return;
```

Comment:
Yes, I was wondering about how to do this. 

It all come from the fact I dont know how to have this included in the pp DQM sequence and not HI sequence (it failed in HI tests). Also speaking as somebody who occasionally has to hack things around to get to run in unusual work flows, having modules which fail gracefully when their inputs are not present is kinda nice. However I admit thats a very weak reason.

So how can I detect that gedGsfElectrons are not present (either in the input file or a producer to make it) or disable them from HI DQM?